File: #100daysofcodewithGFG/Median of 2 Sorted Arrays.py
class Solution:
    def MedianOfArrays(self, array1, array2):
            #code here
            import math
            arr = []
            i = 0
            j = 0
            while i<len(array1) and j<len(array2):
                if array1[i]<array2[j]:
                    arr.append(array1[i])
                    i+=1
                else:
                    arr.append(array2[j])
                    j+=1
            while i<len(array1):
                arr.append(array1[i])
                i+=1
            while j<len(array2):
                arr.append(array2[j])
                j+=1
            mid = len(arr)//2
            if len(arr)%2!=0:
                return arr[mid]
            else:
                ans = (arr[mid]+arr[mid-1])/2
                if math.floor(ans) == ans:
                    return int(ans)
                return ans

File: Problem of the day/Median of 2 sorted arrays.cpp
class Solution{
    public:
    double MedianOfArrays(vector<int>& array1, vector<int>& array2)
    {
        // Your code goes here
        int i = 0;
        int j = 0;
        vector<int> v;
        while (i<array1.size() && j<array2.size()){
            if(array1[i]<array2[j]){
                v.push_back(array1[i]);
                i++;
            }
            else{
                v.push_back(array2[j]);
                j++;
            }
        }
        while(i<array1.size()){
            v.push_back(array1[i]);
            i++;
        }
        while(j<array2.size()){
            v.push_back(array2[j]);
            j++;
        }
        if (v.size()%2 !=0){
            return (double)v[v.size()/2];
        }
        else{
            return (double)(v[v.size()/2]+v[v.size()/2-1])/2;
        }
    
    }
};
